src/typescript/providers/src/wix/wix.provider.ts

```diff
@@ -0,0 +1,171 @@
+import {CustomError} from 'ddns-base';
+import isFQDN from 'validator/es/lib/isFQDN';
+
+import type {Provider, UpdateIpInput} from '../provider.interface';
+import type {WixProviderConfig} from './config.interface';
+
+export class WixProvider implements Provider {
+
+    constructor(
+        private readonly config: WixProviderConfig,
+    ) {
+    }
+
+    async updateIp(input: UpdateIpInput): Promise<void> {
+        const wixApiKey = input.getVariable(this.config.apiKey);
+        if (wixApiKey == null) {
+            throw new Error('Wix provider: apiKey not provided');
+        }
+
+        const wixAccountId = input.getVariable(this.config.accountId);
+        if (wixAccountId == null) {
+            throw new Error('Wix provider: accountId not provided');
+        }
+
+        const wixDnsZone = input.getVariable(this.config.dnsZone);
+        if (wixDnsZone == null) {
+            throw new Error('Wix provider: dnsZone not provided');
+        }
+
+        const recordType = input.ipVersion === 'IPv4' ? 'A' : 'AAAA';
+
+        const hosts = Array.from(new Set([
+            ...this.config.hosts ?? [],
+            ...this.config.useHostsFromRequest === true ? input.requestHosts : [],
+        ]));
+
+        for (const wixHost of hosts) {
+            const host = typeof wixHost === 'string' ? wixHost : wixHost.name;
+            if (host !== wixDnsZone && !host.endsWith(`.${wixDnsZone}`)) {
+                throw new CustomError({
+                    message: `Wix provider: host ${host} should end in ${wixDnsZone}`,
+                });
+            }
+        }
+
+        const getDnsZoneResponse = await this.wixRequest(
+            `domains/v1/dns-zones/${wixDnsZone}`,
```

Comment:
Add comment linking to <https://dev.wix.com/docs/rest/account-level/domains/domain-dns/get-dns-zone>